leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/cli/interactive/InteractiveCommands.java

```diff
@@ -245,25 +255,55 @@ public void run() {
             for (final ServerIdentity server : registeredServers.values()) {
                 LOG.info("Sending Data to {} using {}.", server, contentFormat);
-                parent.client.sendData(server, contentFormat, paths, 2000, new ResponseCallback<SendResponse>() {
-                    @Override
-                    public void onResponse(SendResponse response) {
-                        if (response.isSuccess())
-                            LOG.info("Data sent successfully to {} [{}].", server, response.getCode());
-                        else
-                            LOG.info("Send data to {} failed [{}] : {}.", server, response.getCode(),
-                                    response.getErrorMessage() == null ? "" : response.getErrorMessage());
-                    }
-                }, new ErrorCallback() {
-                    @Override
-                    public void onError(Exception e) {
-                        LOG.warn("Unable to send data to {}.", server, e);
-                    }
-                });
+                int timeoutInMs = 2000;
+                ResponseCallback<SendResponse> responseCallback = (response) -> {
+                    if (response.isSuccess())
+                        LOG.info("Data sent successfully to {} [{}].", server, response.getCode());
+                    else
+                        LOG.info("Send data to {} failed [{}] : {}.", server, response.getCode(),
+                                response.getErrorMessage() == null ? "" : response.getErrorMessage());
+                };
+                ErrorCallback errorCallback = (e) -> LOG.warn("Unable to send data to {}.", server, e);
+                if (senderNames != null) {
+                    // for now noFlush is always false, but we can change that
+                    DataSenderManager dataSenderManager = parent.client.getDataSenderManager();
+                    senderNames.stream().map(sender -> dataSenderManager.getDataSender(sender, ManualDataSender.class))
+                            .forEach(sender -> sender.sendCollectedData(server, contentFormat, timeoutInMs, false));
+                } else {
+                    parent.client.sendData(server, contentFormat, paths, timeoutInMs, responseCallback, errorCallback);
+                }
             }
         }
     }
 
+    /**
+     * A command to collect data.
+     */
+    @Command(name = "collect",
+             description = "Collect data to send it later with 'send' command",
+             headerHeading = "%n",
+             footer = "")
+    static class CollectCommand implements Runnable {
+
+        @Parameters(description = "Paths of data to collect.", converter = StringLwM2mPathConverter.class)
+        private List<String> paths; // TODO we should use LWM2M path
+
+        @Option(names = { "-t", "--to-sender" },
+                required = true,
+                description = { "To which sender the data should be collected" })
+        String senderName;
```

Comment:
From a user point of view I don't think we need 2 sender
I would rather go with only 1 sender and I would not expose his name in the interactive command, so no need to this option

```
Usage:  collect [<paths>...]
Collect data to send it later with 'send' command
      [<paths>...]   Paths of data to collect.
```

I would also change the send command to make sending current value and collected data exclusive
```
Send data to server
      (-cv <paths>... | -cd ) [-c=<contentFormat>]   Paths of data to send.
  -cv, --current-value=<paths>...,
                     Send current value for given path.
  -c, --content-format=<contentFormat>
                     Name (e.g. SENML_JSON) or code (e.g. 110) of Content
                       Format used to send data.
                     Default : SENML_CBOR
   -cd, --collected-data
                     send data collected with collect command.
```

Reply:
> From a user point of view I don't think we need 2 sender
> I would rather go with only 1 sender and I would not expose his name in the interactive command, so no need to this option

OK, but then what's the point of naming them (i.e. mapping them by a `String` in `DataSenderManager`)? I don't see why we would allow having multiple senders if user can only access one (and which one would that be?).

Reply:
There is the **Leshan API** and there is a the **Leshan demo CLI**. 
Leshan demo is just an example of  Leshan API usage.
Making the Leshan API more flexible doesn't not mean that all the flexibility should be available in demo CLI. :slightly_smiling_face: 


Here, we hide the name in CLI but the name is used by demo code to access easily/elegantly to the manual sender.
And if/when we will add the PeriodicSender, name will be also useful to access to one or another.

Reply:
So I'm guessing if we're still planning on using a map for storing senders but we will only have one sender with a hidden name, then we have to use some hard-coded String as a name, right?